File: 1-1000/209_minimum_size_subarray_sum/solution.go
package main

import (
	"fmt"
	"math"
)

func minSubArrayLen(target int, nums []int) int {
	var left, right, currSum int
	var minLength = math.Inf(1)

	for left < len(nums) && right < len(nums) {
		for right < len(nums) && currSum < target {
			currSum += nums[right]
			right++
		}
		for left < right && currSum >= target {
			minLength = min(float64(right-left), minLength)
			currSum -= nums[left]
			left++
		}
	}
	if minLength == math.Inf(1) {
		return 0
	}
	return int(minLength)
}

func main() {
	var nums = []int{2, 3, 1, 2, 4, 3}
	var target int = 7

	fmt.Println(nums)
	fmt.Println(target)
	fmt.Println(minSubArrayLen(target, nums))
}

File: 1-1000/209_minimum_size_subarray_sum/solution.py
def solve(nums: list, target: int):
    min_lens = []

    for i in range(len(nums)):
        curr_len = 1
        sum = nums[i]
        j = i + 1
        while j < len(nums) and sum < target:
            sum += nums[j]
            j += 1
            curr_len += 1
        if sum >= target:
            min_lens.append(curr_len)
    return min(min_lens) if min_lens else 0


print(solve([2, 3, 1, 2, 4, 3], 7))
print(solve([1, 1, 1, 1], 7))
